src/App.js

```diff
@@ -59,42 +61,61 @@ class App extends Component {
         <Route exact path="/login" component={Login} />
         <Route exact path="/signup" component={Signup} />
         <Route exact path="/projects/new" component={NewProject} />
+      </>
+    );
+  };
+
+  constructComponent = (componentName) => {
+    let route = "/projects/p/:hash/";
+    let endPoint = this.components[componentName][1];
+    let queries = this.components[componentName][2];
+
+    let path = route + endPoint + queries;
+    let activeLink = endPoint;
```

Comment:
As these aren't modified at any point wouldn't 'const' be more appropriate than 'let'?

```suggestion
    const activeLink = endPoint;
```

Reply:
```
let` is a signal that the variable may be reassigned, such as a counter in a loop, or a value swap in an algorithm. It also signals that the variable will be used only in the block it’s defined in, which is not always the entire containing function.
```

“JavaScript ES6+: var, let, or const?” by Eric Elliott https://link.medium.com/CNpSPBFtm8

It's all about signalling your intention, this matters as you won't already be the only one writing code. Also it helps when you go back and look at it r yourself after a long time.